### PR TITLE
Handle bit types properly so that UPDATE and DELETE work as expected during replication

### PR DESCRIPTION
**Pull request by deepthi**
This is for #4145. 
Changed how bit literals are handled in vitess so that BIT fields are matched correctly for UPDATE / DELETE. 
Instead of converting them to '\x0...' format, we will convert them to b'...' format.

### CODE REVIEW

File: go/vt/sqlparser/ast.go [hunk 1]
```diff
@@ -2397,7 +2397,7 @@ func ExprFromValue(value sqltypes.Value) (Expr, error) {
 		return NewIntVal(value.ToBytes()), nil
 	case value.IsFloat() || value.Type() == sqltypes.Decimal:
 		return NewFloatVal(value.ToBytes()), nil
-	case value.IsQuoted():
+	case value.IsQuoted() || value.Type() == sqltypes.Bit:
```
Comment (sougou) [inline go/vt/sqlparser/ast.go]: I did a sanity check (based on where this function is called from); It's used to initialize the column with the default value as returned by the `describe` statement:

```
mysql> create table b(id int, bv bit(8) default 'a', primary key(id));
Query OK, 0 rows affected (0.03 sec)

mysql> describe b;
+-------+---------+------+-----+------------+-------+
| Field | Type    | Null | Key | Default    | Extra |
+-------+---------+------+-----+------------+-------+
| id    | int(11) | NO   | PRI | NULL       |       |
| bv    | bit(8)  | YES  |     | b'1100001' |       |
+-------+---------+------+-----+------------+-------+
2 rows in set (0.00 sec)
```

In other words, just parsing this as a string will yield the wrong default value.

For now, I don't think it's worth supporting this here. So, let's just return an error.

File: go/sqltypes/value.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func NewValue(typ querypb.Type, val []byte) (v Value, err error) {
 			return NULL, err
 		}
 		return MakeTrusted(typ, val), nil
-	case IsQuoted(typ) || typ == Null:
+	case IsQuoted(typ) || typ == Bit || typ == Null:
```
[the remark this answers is not in the document]
Comment (sougou) [reply go/sqltypes/value.go]: Also, `MarshalJSON` should use `ToString` (same as quoted) for bit types.

File: go/sqltypes/value.go [hunk 3]
```diff
@@ -333,6 +335,26 @@ func encodeBytesSQL(val []byte, b BinWriter) {
 	b.Write(buf.Bytes())
 }
 
+func encodeBytesSQLBits(val []byte, b BinWriter) {
+	buf := &bytes2.Buffer{}
+	buf.WriteByte('b')
+	buf.WriteByte('\'')
+	for _, ch := range val {
+		// for each byte, create a string representation of the bits
```
[the remark this answers is not in the document]
Comment (deepthi) [reply go/sqltypes/value.go]: Nice! This makes the code more compact and readable too.

**Comment by tirsen** [comment]
YAY!

**Pull request merged**